src/imageio/format/j2k.c

```diff
@@ -395,9 +395,15 @@ int write_image(dt_imageio_module_data_t *j2k_tmp, const char *filename, const v
 //        }
 //        break;
       case 12:
-        for(int i = 0; i < w * h; i++)
+#ifdef _OPENMP
+#pragma omp parallel for simd default(none) \
```

Comment:
If we use `simd` here, wouldn't we possibly write/read data out of bounds?

Reply:
Should I just unroll the inner loop then?

Reply:
i am not the compiler expert but i would just use `omp parallel for default(none)` , i think we had issues like this before.

Reply:
The `simd` shouldn't cause out-of-bounds issues; it will alter how the compiler divides work among threads to keep chunk boundaries aligned to the vector size.